File: leetcode/1easy/위성원/121.ts
function maxProfit(prices: number[]): number {
  const arr = prices
    .map((p, index) => ({
      p,
      index,
    }))
    .sort((a, b) => {
      if (a.p === b.p) {
        return a.index - b.index;
      }

      return a.p - b.p;
    });

  let max = 0;
  for (let i = 0; i < arr.length; i++) {
    const v = arr[i];
    for (let j = arr.length - 1; j > i; j--) {
      const vj = arr[j];
      if (max > vj.p - v.p) {
        break;
      }

      if (vj.index > v.index) {
        max = vj.p - v.p;
      }
    }
  }

  return max;
}

File: leetcode/1easy/박은비/121.swift
<<<<<<< HEAD
class Leetcode.이재훈.이재훈.Solution {
=======
class Leetcode.이재훈.Solution {
>>>>>>> ab7f46f73d8f18bcd38c5cf06ecabae64f53a424
    func maxProfit(_ prices: [Int]) -> Int {
        var left: Int = prices[0]
        var right: Int = prices[0]
        var maxProfit = 0

        for i in 1..<prices.count {
            
            if left>prices[i] {
                left = prices[i]
                right = prices[i]
            } else {
                right = prices[i]
                maxProfit = max(maxProfit, right-left)
            }

        }

        return maxProfit
    }
}
